web/src/app/util/useWidth.js

```diff
@@ -0,0 +1,20 @@
+import { useTheme } from '@material-ui/core/styles'
+import useMediaQuery from '@material-ui/core/useMediaQuery'
+
+/**
+ * Only works because the number of with default mui breakpoints.
+ * It will break once you change the number of breakpoints.
+ * See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level
+ *
+ * https://material-ui.com/components/use-media-query/#migrating-from-withwidth
```

Comment:
Move this to the top with a note indicating this was copied directly from the source.